### PR TITLE
Add ProcessEvent for OpenDebugAD7

### PR DESCRIPTION
**Pull request by WardenGnaw**
This PR adds the process name, pid, startMethod and pointerSize for ProcessEvent.

Also updating VSCodeDebugProtocol.

### CODE REVIEW

File: src/Microsoft.VisualStudio.Debugger.Interop.DAP/Microsoft.VisualStudio.Debugger.Interop.DAP.cs [hunk 1]
```diff
@@ -35,4 +35,21 @@ public interface IDebugExpressionDAP
         [PreserveSig]
         int EvaluateSync([In] enum_EVALFLAGS dwFlags, [In] DAPEvalFlags dapFlags, [In] uint dwTimeout, [In][MarshalAs(UnmanagedType.Interface)] IDebugEventCallback2 pExprCallback, [Out, MarshalAs(UnmanagedType.Interface)] out IDebugProperty2 ppResult);
     }
+
+    /// <summary>
+    ///
```
Comment (WardenGnaw) [inline src/Microsoft.VisualStudio.Debugger.Interop.DAP/Microsoft.VisualStudio.Debugger.Interop.DAP.cs:40]: nit: Add information on this new Interface

File: src/Microsoft.VisualStudio.Debugger.Interop.DAP/Microsoft.VisualStudio.Debugger.Interop.DAP.cs [hunk 2]
```diff
@@ -35,4 +35,21 @@ public interface IDebugExpressionDAP
...
+    /// 
+    /// </summary>
+    [ComImport()]
+    [ComVisible(true)]
+    [Guid("1615C23F-2BA4-4018-A135-E8B889B01F3E")]
+    [InterfaceType(1)]
+    public interface IDebugProgramDAP
+    {
+        /// <summary>
+        ///
```
Comment (WardenGnaw) [inline src/Microsoft.VisualStudio.Debugger.Interop.DAP/Microsoft.VisualStudio.Debugger.Interop.DAP.cs:49]: Describe what this API is for.

Comment (gregg-miskelly) [inline src/Microsoft.VisualStudio.Debugger.Interop.DAP/Microsoft.VisualStudio.Debugger.Interop.DAP.cs:49]: Add something for the description? And be clear that the pointer size should be in bits (ex: typically `64` or `32`).

File: src/OpenDebugAD7/AD7DebugSession.cs [hunk 3]
```diff
@@ -2472,6 +2473,38 @@ public void HandleIDebugMessageEvent2(IDebugEngine2 pEngine, IDebugProcess2 pPro
...
+        {
+            IDebugProcessInfoUpdatedEvent158 debugProcessInfoUpdated = pEvent as IDebugProcessInfoUpdatedEvent158;
+
+            if (debugProcessInfoUpdated != null && 
+                debugProcessInfoUpdated.GetUpdatedProcessInfo(out string name, out uint systemProcessId) == HRConstants.S_OK)
+            {
+                // Send ProcessEvent to Client
+                ProcessEvent processEvent = new ProcessEvent();
+                processEvent.Name = name;
+                processEvent.SystemProcessId = (int)systemProcessId;
```
Comment (WardenGnaw) [inline src/OpenDebugAD7/AD7DebugSession.cs:2486]: @pieandcakes Should we save `systemProcessId` in AD7DebugSession? This is the actual debuggee process reported from GDB.

File: src/OpenDebugAD7/AD7DebugSession.cs [hunk 4]
```diff
@@ -2472,6 +2473,38 @@ public void HandleIDebugMessageEvent2(IDebugEngine2 pEngine, IDebugProcess2 pPro
...
+        public void HandleIDebugProcessInfoUpdatedEvent158(IDebugEngine2 pEngine, IDebugProcess2 pProcess, IDebugProgram2 pProgram, IDebugThread2 pThread, IDebugEvent2 pEvent)
+        {
+            IDebugProcessInfoUpdatedEvent158 debugProcessInfoUpdated = pEvent as IDebugProcessInfoUpdatedEvent158;
+
+            if (debugProcessInfoUpdated != null && 
+                debugProcessInfoUpdated.GetUpdatedProcessInfo(out string name, out uint systemProcessId) == HRConstants.S_OK)
+            {
+                // Send ProcessEvent to Client
+                ProcessEvent processEvent = new ProcessEvent();
+                processEvent.Name = name;
```
Comment (WardenGnaw) [inline src/OpenDebugAD7/AD7DebugSession.cs:2485]: And update `m_processName`?

File: src/OpenDebugAD7/AD7DebugSession.cs (hunk 3 above)
Comment (WardenGnaw) [reply src/OpenDebugAD7/AD7DebugSession.cs:2486]: I'll do this in another PR, this can help Tracepoints evaluate `$PROCESSID`

**Pull request merged**